flask_restful/__init__.py

```diff
@@ -284,6 +285,17 @@ def handle_error(self, e):
         data = getattr(e, 'data', error_data(code))
         headers = {}
 
+        error_cls_name = type(e).__name__
+        if error_cls_name in self.errors:
+            custom_data = self.errors.get(error_cls_name, {})
+            code = custom_data.get('status', 500)
+            data.update(custom_data)
+
+        for typecheck, handler in self.errorhandlers:
+            if isinstance(e, typecheck):
+                data, code, headers = unpack(handler(e))
+                break
+
         if code >= 500:
```

Comment:
These checks will probably need to be removed and replaced with default errorhandler functions that are set up when an API object is created. It looks like if you were to add an errorhandler to catch `BadRequest`, your custom resposne would be overridden by these checks.